backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -536,30 +617,89 @@ public FutureExperimentRunDAO(
               return localQueryContext;
             },
             executor);
-
-    final var futurePredicatesContext =
-        predicatesHandler.processPredicates(request.getPredicatesList(), executor);
-
-    final var futureSortingContext =
-        sortingHandler.processSort(request.getSortKey(), request.getAscending());
-
-    final var futureProjectIds =
-        getAllowedProjects(ModelDBActionEnum.ModelDBServiceActions.READ)
-            .thenApply(
-                projIds ->
-                    new QueryFilterContext()
-                        .addCondition("experiment_run.project_id in (<authz_project_ids>)")
-                        .addBind(q -> q.bindList("authz_project_ids", projIds)),
-                executor);
+    queryFilterContextList.add(futureLocalContext);
+
+    // futurePredicatesContext
+    queryFilterContextList.add(
+        predicatesHandler.processPredicates(request.getPredicatesList(), executor));
+
+    // futureSortingContext
+    queryFilterContextList.add(
+        sortingHandler.processSort(request.getSortKey(), request.getAscending()));
+
+    // futureProjectIds based on workspace
+    if (request.getWorkspaceName().isEmpty()) {
```

Comment:
This whole logic is way too big to be inlined in this method. Extract it to a different method.

Reply:
extracted in separate method